lib/DBDish/Pg/ErrorHandling.pm6

```diff
@@ -0,0 +1,38 @@
+use v6;
+need DBDish::ErrorHandling;
+
+package X::DBDish {
+    class DBError::Pg is X::DBDish::DBError {
+        has $.sqlstate is required;
+        has $.message-detail;
+        has $.message-hint;
+        has $.context;
+        has $.type;
+        has $.type-localized;
+
+        has $.statement-position;
+        has $.internal-position;
+        has $.internal-query;
+
+        has $.schema;
+        has $.table;
+        has $.column;
+        has $.datatype;
+        has $.constraint;
+
+        has $.source-file;
+        has $.source-line;
+        has $.source-function;
+
+        # Errors which should cause a retry loop within the calling application include:
+        #  - Class 08\w{3}: All connection exceptions (possible network issues)
+        #  - 40001 serialization_failure
+        #  - 40P01 deadlock_detected
+        #  - Class 57\w{3}: Operator Intervention (early/forced connection termination)
+        #  - 72000 snapshot_too_old
+        method is-temporary {
+           return True when ($.sqlstate ~~ /^ '08'<[alnum]> ** 3 | '40001'| '40P01' | '57'<[alnum]> ** 3 | '72000' $/ );
```

Comment:
This may just be a matter of style, but these two lines could also be written as just:
`so $.sqlstate ~~ /^ '08'<[alnum]> ** 3 | '40001'| '40P01' | '57'<[alnum]> ** 3 | '72000' $/`